src/Shared/HttpSys/RequestProcessing/RequestHeaders.Generated.tt

```diff
@@ -213,5 +212,28 @@ namespace Microsoft.AspNetCore.HttpSys.Internal
             }
 <# } #>
         }
+
+        private static ReadOnlySpan<byte> HeaderKeys => new byte[]
+        {
+<# foreach(var prop in props) { #>
+            (byte)<#=prop.ID#>,
+<# } #>
+        };
+
+        private string GetHeaderKeyName(HttpSysRequestHeader header) => header switch
+        {
+<# foreach(var prop in props) { #>
+            <#=prop.ID#> => HeaderNames.<#=prop.Name#>,
+<# } #>
+            _ => throw new NotImplementedException()
+        };
+
+        private bool HasKnownHeader(HttpSysRequestHeader header) => header switch
+        {
+<# foreach(var prop in props) { #>
+            <#=prop.ID#> => <#=IsRead(prop.Index)#> ? _<#=prop.Name#>.Count > 0 : _requestMemoryBlob.HasKnownHeader(<#=prop.ID#>),
```

Comment:
Note, that I am using `_<#=prop.Name#>.Count > 0` here (with underscore prefix), and previously _PropertiesTryRemove_ method has been using without the underscore:

```csharp
(<#=IsRead(prop.Index)#> && <#=prop.Name#>.Count > 0)
  || (!<#=IsRead(prop.Index)#> && HasKnownHeader(<#=prop.ID#>))
```